### PR TITLE
Task 3: Update / Edit Todo

### PR DESCRIPTION
**Pull request by aka-wtag**
Objectives:
- [x] Beside each todo, there is another new button named Edit.
- [x] Clicking the Edit todo button will the turn the current Todo text content into an input box.
- [x] When in Edit Mode, The text of Add button is changed from Edit to Update.
- [x] If a user clicked the edit button, but don’t want any changes, then there should be a button named cancel to go back to the previous state.

[T3.webm](https://github.com/aks-wtag/todo-app/assets/152831915/28d59743-7f29-4e69-9dcb-aa343e04dd54)

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -37,16 +37,68 @@ const deleteTaskHandler = (event) => {
     taskList = taskList.filter((task) => task.id!==taskId);
 };
 
+const updateTaskEditHandler = (task, $taskDetails, $inputField, $editButton, $updateButton, $cancelButton) => {
+    task.title = $inputField.value;
+    
+    taskList.map((t) => {
```
Comment (njm-wtag) [inline js/script.js:43]: It's not recommended to use meaningless variable name rather we can use `todo` or `singleTask` insted of `t`

**Review by nmi-wtag — CHANGES_REQUESTED** [review]

File: js/script.js [hunk 2]
```diff
@@ -37,16 +37,68 @@ const deleteTaskHandler = (event) => {
...
+        if(t.id==task.id){
```
Comment (njm-wtag) [inline js/script.js:44]:
```suggestion
        if(t.id===task.id){
```

File: js/script.js [hunk 3]
```diff
@@ -37,16 +37,68 @@ const deleteTaskHandler = (event) => {
     taskList = taskList.filter((task) => task.id!==taskId);
 };
 
+const updateTaskEditHandler = (task, $taskDetails, $inputField, $editButton, $updateButton, $cancelButton) => {
+    task.title = $inputField.value;
```
Comment (njm-wtag) [inline js/script.js:41]: What happen if input value is empty? We can check error

File: js/script.js (hunk 1 above)
Comment (njm-wtag) [inline js/script.js:43]: Why are we using `map` here? As its return a new array. We can use `find` instead.

**Review by nmi-wtag — APPROVED** [review]

File: js/script.js [hunk 4]
```diff
@@ -41,16 +39,68 @@ const deleteTaskHandler = (event) => {
     tasks = tasks.filter((task) => task.id !== taskId);
 };
 
+const updateTaskEditHandler = (task, $taskDetails, $inputField, $editButton, $updateButton, $cancelButton) => {
+    const taskTitle = sanitizeInput($inputField.value);
+    
+    if (!taskTitle) {
+        showErrorMessage("Task Title must be provided for edit");
+        return;
```
Comment (hhu-wtag) [inline js/script.js:47]:
```suggestion
        showErrorMessage("Task Title must be provided for edit");
        
        return;
```

File: js/utility.js
Comment (hhu-wtag) [inline js/utility.js:1]: add eol

**Review by hhu-wtag — APPROVED** [review]

**Pull request closed without merging**